src/Day-2/RegistrationForm.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import './style.css';
+
+export default class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: {
+        value: '',
+        error: ''
+      },
+      lastName: {
+        value: '',
+        error: ''
+      },
+      email: {
+        value: '',
+        error: ''
+      },
+      contactNo: {
+        value: '',
+        error: ''
+      },
+      password: {
+        value: '',
+        error: ''
+      },
+      confirmPassword: {
+        value: '',
+        error: ''
+      },
+      hasError: true
+    };
+
+    this.updateForm = this.updateForm.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  updateForm(e) {
+    let initialData = this.state;
+    initialData[e.target.name] = e.target.value;
+    this.setState(initialData);
+  }
+
+  submitForm() {
+    let re =  /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    let hasError = false;
+
+    if(this.state.firstName.value === '') {
+      this.setState(Object.assign(this.state.firstName, {error: 'First name cannot be blank'}));
+      hasError = true;
+    }
+    if(this.state.lastName.value === '') {
+      this.setState(Object.assign(this.state.lastName, {error: 'Last name cannot be blank'}));
+      hasError = true;
+    }
+    if((this.state.firstName.value !== this.state.lastName.value)) {
+      this.setState(Object.assign(this.state.firstName, {error: 'First name is not equal to last name'}));
+      this.setState(Object.assign(this.state.lastName, {error: 'Last name is not equal to first name'}));
+      hasError = true;
+    }
+
+    if(this.state.password.value !== this.state.confirmPassword.value) {
+      this.setState(Object.assign(this.state.password, {error: 'Password is not equal to confirm password'}));
+      hasError = true;
+    }
+
+    if(!re.test(this.state.email.value)) {
+      this.setState(Object.assign(this.state.email, {error: 'Email is not valid'}));
+      hasError = true;
+    }
+
+    if(this.state.contactNo.toString().length !== 10) {
+      this.setState(Object.assign(this.state.contactNo, {error: 'Phone no. should be of 10 digits'}));
+      hasError = true;
+    }
+
+    this.setState({
+      hasError: hasError
+    })
+  }
+
+  render() {
+    return (
+      <div>
+        <h2>Form</h2>
+        <table>
+          <tr>
+            <td>FirstName</td>
+            <td>
+              <input type="text" name="firstName" value={this.state.firstName.value} onChange={this.updateForm}/>
+              {this.state.firstName.error === '' ? '' : <p className="errorString">{this.state.firstName.error}</p>}
```

Comment:
could have just used
`this.state.firstName.error ? '' : <p className="errorString">{this.state.firstName.error}</p>`